src/librustdoc/html/sources.rs

```diff
@@ -99,16 +99,15 @@ impl<'a> SourceCollector<'a> {
             href.push('/');
         });
         self.scx.ensure_dir(&cur)?;
-        let mut fname = p.file_name().expect("source has no filename").to_os_string();
-        fname.push(".html");
+
+        let src_fname =
+            String::from(p.file_name().expect("source has no filename").to_string_lossy());
+        let fname = OsString::from(src_fname.clone() + ".html");
         cur.push(&fname);
         href.push_str(&fname.to_string_lossy());
 
-        let title = format!(
-            "{} -- source",
-            cur.file_name().expect("failed to get file name").to_string_lossy()
-        );
-        let desc = format!("Source to the Rust file `{}`.", filename);
+        let title = format!("{} – source", src_fname,);
```

Comment:
It *might* be better to just use a plain dash (-) rather than a Unicode en dash (–) since I think that's what other page titles use.

Reply:
I would prefer to use `-` for now so it's consistent; if you want to use `–` then let's make a separate PR making all of the pages use `–` at once.